egs/wsj/s5/utils/validate_lang.pl

```diff
@@ -121,6 +129,7 @@ sub check_allowed_whitespace {
   print "Usage: $0 [options] <lang_directory>\n";
   print "e.g.:  $0 data/lang\n";
   print "Options:\n";
+  print " --subword <separator>                    (this flag indicate that the lang is for subword)";
```

Comment:
I think it's better if you detect this automatically, again based on the fact that the word-boundary stuff is moved.
You can have your script create a file $lang/subword_separator.txt where you can put the separator.
I guess you could use the existence of that file as the clue that this is a sub-word lexicon.